### PR TITLE
Fix issue where kubernetesDashboard params weren't being added despite e enabling the dashboard addon

### LINKED ISSUE #4078: k8s cluster upgrade fails with template error
**Is this a request for help?**:
yes
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
ISSUE
---

**What version of acs-engine?**:
0.24.0
---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes 1.10.7

**What happened**:
When trying to upgrade Cluster to 1.10.9 the following error is thrown:

> INFO[0000] validating...                                
> INFO[0004] Name suffix: 12658618                        
> INFO[0004] Gathering agent pool names...                
> INFO[0005] VM k8s-dynamic-12658618-vmss_0 in VMSS k8s-dynamic-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-dynamic-12658618-vmss_1 in VMSS k8s-dynamic-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-dynamic-12658618-vmss_2 in VMSS k8s-dynamic-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-dynamic-12658618-vmss_3 in VMSS k8s-dynamic-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-graph-12658618-vmss_0 in VMSS k8s-graph-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_0 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
>INFO[0005] VM k8s-static-12658618-vmss_1 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_2 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_3 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_4 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_5 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_6 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_7 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_8 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_9 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_10 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_11 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_12 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_13 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_14 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_15 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_16 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_17 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_18 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_19 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] VM k8s-static-12658618-vmss_20 in VMSS k8s-static-12658618-vmss has a current tag of Kubernetes:1.10.7 and a desired tag of Kubernetes:1.10.9. Upgrading this node.
> 
> INFO[0005] Master VM name: k8s-master-12658618-0, orchestrator: Kubernetes:1.10.7 (MasterVMs)
> 
> INFO[0005] Upgrading to Kubernetes version 1.10.9
>      
> INFO[0005] Master nodes StorageProfile: ManagedDisks    
> INFO[0005] Prepping master nodes for upgrade...         
> INFO[0005] Resource count before running NormalizeResourcesForK8sMasterUpgrade: 10 
> INFO[0005] Evaluating if agent pool: master, resource: [concat(variables('masterVMNamePrefix'), copyIndex(variables('masterOffset')))] needs to be removed 
> INFO[0005] Evaluating if extension: [concat(variables('masterVMNamePrefix'), copyIndex(variables('masterOffset')),'/cse', '-master-', copyIndex(variables('masterOffset')))] needs to be removed 
> INFO[0005] Evaluating if extension: [concat(variables('masterVMNamePrefix'), copyIndex(variables('masterOffset')), '/computeAksLinuxBilling')] needs to be removed 
> INFO[0005] Resource count after running NormalizeResourcesForK8sMasterUpgrade: 10 
> INFO[0005] Total expected master count: 1               
> INFO[0005] Master nodes that need to be upgraded: 1     
> INFO[0005] Master nodes that have been upgraded: 0      
> INFO[0005] Starting upgrade of master nodes...          
> INFO[0005] masterNodesInCluster: 1                      
> INFO[0005] Upgrading Master VM: k8s-master-12658618-0   
> INFO[0005] fetching VM: dev_004_QKNOWS_K8s/k8s-master-12658618-0 
> INFO[0006] found nic name for VM (dev_004_QKNOWS_K8s/k8s-master-12658618-0): k8s-master-12658618-nic-0 
> INFO[0006] deleting VM: dev_004_QKNOWS_K8s/k8s-master-12658618-0 
> INFO[0006] waiting for vm deletion: dev_004_QKNOWS_K8s/k8s-master-12658618-0 
> INFO[0102] deleting nic: dev_004_QKNOWS_K8s/k8s-master-12658618-nic-0 
> INFO[0102] waiting for nic deletion: dev_004_QKNOWS_K8s/k8s-master-12658618-nic-0 
> INFO[0102] deleting managed disk: dev_004_QKNOWS_K8s/k8s-master-12658618-0_OsDisk_1_08bcf62084bd4234858234979722d2ba 
> INFO[0163] Master offset: 0
>                            
> INFO[0163] Master pool set count to: 1 temporarily during upgrade...
>  
> INFO[0163] Starting ARM Deployment (master-18-10-18T20.58.38-194173334). This will take some time... 
> INFO[0164] Error creating upgraded master VM: k8s-master-12658618-0 
> FATA[0164] Error upgrading cluster: resources.DeploymentsClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="InvalidTemplate" Message="Deployment template validation failed: 'The value for the template parameter 'kubernetesDashboardCPULimit' at line '1' and column '16921' is not provided. Please see https://aka.ms/arm-deploy/#parameter-file for usage details.'."

This unfortunately happens after the Master node was already deleted.

**What you expected to happen**:
Cluster upgrade to complete successfully

**How to reproduce it** (as minimally and precisely as possible):
Initial setup of Cluster with acs-engine 0.21.2:
- Kubernetes 1.10.7
- Private cluster
- RBAC enabled
- Calico Networkpolicy
- Disable addons:
   - blobfuse-flexvolume
   - smb-flexvolume
- 1 Master
- 3 Agentpools as vmss

Command used to upgrade:
acs-engine upgrade --subscription-id <subscription> --resource-group <rg-name>--location northeurope --deployment-dir deployment-20181004_104850/arm-deploy --upgrade-version 1.10.9 --auth-method client_secret --client-id <clientid> --client-secret <secret> 

**Anything else we need to know**:
Cluster could be brought back up with using acs-engine 0.21.2 and upgrading to original k8s version 1.10.7.

### PR DESCRIPTION
**Pull request by tariq1890**
This PR fixes a regression introduced in release v0.23.0 . Fixes #4078

### CODE REVIEW

File: pkg/acsengine/params_k8s.go [hunk 1]
```diff
@@ -148,7 +148,7 @@ func assignKubernetesParameters(properties *api.Properties, parametersMap params
 			}
 			if kubernetesConfig.IsDashboardEnabled() {
 				dashboardAddon := kubernetesConfig.GetAddonByName(DefaultDashboardAddonName)
-				dashboardAddon.GetAddonContainersIndexByName(DefaultDashboardAddonName)
+				c := dashboardAddon.GetAddonContainersIndexByName(DefaultDashboardAddonName)
```
Comment (jackfrancis) [inline pkg/acsengine/params_k8s.go]: gah

can we change the name of var `c` above? If that wasn't `c` then we wouldn't gotten a compile time error and this bug would never have happened

**Comment by tariq1890** [comment]
Sure. I can do that for other add-ons as well. In this instance, ill rename it to dashboardIndex

**Comment by tariq1890** [comment]
That okay?

File: pkg/acsengine/params_k8s.go (hunk 1 above)
Comment (tariq1890) [reply pkg/acsengine/params_k8s.go]: Sure. I can do that for other add-ons as well. In this instance, ill rename it to dashboardIn

Comment (jackfrancis) [reply pkg/acsengine/params_k8s.go]: 👍

**Comment by tariq1890** [comment]
@jackfrancis Review comments addressed.

**Pull request merged**